packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -281,43 +353,123 @@ async function getAbapService(config: AbapDeployConfig, logger: Logger): Promise
     return service;
 }
 
+/**
+ * Updates the config (ui5-deploy.yaml) file with the new transport request.
+ *
+ * @param config - deployment configuration
+ * @param logger - reference to the logger instance
+ * @param configPath - path to the config file
+ * @returns Promise<void>
+ */
+async function updateYaml(config: AbapDeployConfig, logger: Logger, configPath?: string): Promise<void> {
+    const deployYamlPath = configPath ?? join(process.cwd(), 'ui5-deploy.yaml');
+    if (existsSync(deployYamlPath)) {
+        try {
+            const fs = create(createStorage());
+            const ui5DeployConfig = await UI5Config.newInstance(fs.read(deployYamlPath));
+            const abapTarget = {
+                ...config.target,
+                scp: config.target.cloud
+            } as AbapTarget;
+            ui5DeployConfig.updateAbapDeployTask(abapTarget, config.app);
+            fs.write(deployYamlPath, ui5DeployConfig.toString());
+            return new Promise((resolve) => {
+                fs.commit(resolve);
+                logger.info('Updated the ui5-deploy.yaml with the new transport request');
```

Comment:
Even though Fiori tools uses ui5-deploy.yaml, there is nothing stopping a developer from using any filename when deploying to ABAP.